config/config_remote_write.go

```diff
@@ -0,0 +1,162 @@
+func (t RemoteWriteProtoTypes) ServerAcceptHeaderValue() string {
+	// TODO(bwplotka): Consider implementing an optional quality factor.
+	ret := make([]string, 0, len(t))
+	for _, typ := range t {
+		ret = append(ret, RemoteWriteContentTypeHeaders[typ])
+	}
+	return strings.Join(ret, ",")
+}
+
+var (
+	RemoteWriteProtoTypeV1        RemoteWriteProtoType = "v1.WriteRequest"
+	RemoteWriteProtoTypeV2        RemoteWriteProtoType = "v2.WriteRequest"
+	RemoteWriteContentTypeHeaders                      = map[RemoteWriteProtoType]string{
+		RemoteWriteProtoTypeV1: "application/x-protobuf", // Also application/x-protobuf; proto=prometheus.WriteRequest but simplified for compatibility with 1.x spec.
+		RemoteWriteProtoTypeV2: "application/x-protobuf;proto=io.prometheus.remote.write.v2.WriteRequest",
+	}
+
+	// DefaultRemoteWriteProtoTypes is the set of remote write protobuf types that will be
+	// preferred by the remote write client.
+	DefaultRemoteWriteProtoTypes = RemoteWriteProtoTypes{
+		RemoteWriteProtoTypeV1,
+		RemoteWriteProtoTypeV2,
+	}
+)
+
+// validateRemoteWriteProtoTypes return errors if we see problems with rw protobuf types in
+// the Prometheus configuration.
+func validateRemoteWriteProtoTypes(ts []RemoteWriteProtoType) error {
+	if len(ts) == 0 {
+		return errors.New("protobuf_types cannot be empty")
+	}
+	dups := map[string]struct{}{}
+	for _, t := range ts {
+		if _, ok := dups[strings.ToLower(string(t))]; ok {
+			return fmt.Errorf("duplicated protobuf types in protobuf_types, got %v", ts)
+		}
+		if err := t.Validate(); err != nil {
+			return fmt.Errorf("protobuf_types: %w", err)
+		}
+		dups[strings.ToLower(string(t))] = struct{}{}
+	}
+	return nil
+}
+
+// RemoteWriteCompression represents the supported compressions for the remote write.
+type RemoteWriteCompression string
+
+// Validate returns error if the given protobuf type is not supported.
+func (s RemoteWriteCompression) Validate() error {
+	if _, ok := RemoteWriteContentEncodingHeaders[s]; !ok {
+		return fmt.Errorf("unknown remote write protobuf type %v, supported: %v",
+			s, func() (ret []string) {
+				for k := range RemoteWriteContentEncodingHeaders {
+					ret = append(ret, string(k))
+				}
+				sort.Strings(ret)
+				return ret
+			}())
+	}
+	return nil
+}
+
+type RemoteWriteCompressions []RemoteWriteCompression
+
+func (cs RemoteWriteCompressions) Strings() []string {
+	ret := make([]string, 0, len(cs))
+	for _, c := range cs {
+		ret = append(ret, string(c))
+	}
+	return ret
+}
+
+func (cs RemoteWriteCompressions) String() string {
+	return strings.Join(cs.Strings(), ",")
+}
+
+// ServerAcceptEncodingHeaderValue returns server Accept-Encoding header value for
+// given list of compressions as per RFC 9110 https://www.rfc-editor.org/rfc/rfc9110.html#name-accept-encoding
+func (cs RemoteWriteCompressions) ServerAcceptEncodingHeaderValue() string {
+	// TODO(bwplotka): Consider implementing an optional quality factor.
```

Comment:
by not doing so yet we're just interpreting the order as the preference ordering, right?

Reply:
Yes, which is part of RFC 9110 spec, q is optional and allow to tell "weight" of the ordering.